analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Loops.cs

```diff
@@ -25,6 +25,92 @@ namespace SonarAnalyzer.Test.SymbolicExecution.Roslyn;
+    }
+
+    [TestMethod]
+    public void Loops_IsInLoop_ForEach()
+    {
+        var code = """
+            Console.WriteLine("Before loop");
+            foreach (var i in items)
+            {
+                Console.WriteLine("Inside loop");
+            }
+            Console.WriteLine("After loop");
+            """;
+        var validator = SETestContext.CreateCS(code, "int[] items").Validator;
+        validator.PostProcessed.GroupBy(x => x.Block.Ordinal).Should().HaveCount(7);
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 3 or 4).Should().AllSatisfy(x => x.IsInLoop.Should().BeTrue());
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 1 or 2 or 5 or 6 or 8).Should().AllSatisfy(x => x.IsInLoop.Should().BeFalse());
+    }
+
+    [TestMethod]
+    public void Loops_IsInLoop_While()
+    {
+        var code = """
+            Console.WriteLine("Before loop");
+            while (Condition)
+            {
+                Console.WriteLine("Inside loop");
+            }
+            Console.WriteLine("After loop");
+            """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.PostProcessed.GroupBy(x => x.Block.Ordinal).Should().HaveCount(4);
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 2 or 3).Should().AllSatisfy(x => x.IsInLoop.Should().BeTrue());
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 1 or 4).Should().AllSatisfy(x => x.IsInLoop.Should().BeFalse());
+    }
+
+    [TestMethod]
+    public void Loops_IsInLoop_DoWhile()
+    {
+        var code = """
+            Console.WriteLine("Before loop");
+            do
+            {
+                Console.WriteLine("Inside loop");
+            }
+            while (Condition);
+            Console.WriteLine("After loop");
+            """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.PostProcessed.GroupBy(x => x.Block.Ordinal).Should().HaveCount(3);
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 2).Should().AllSatisfy(x => x.IsInLoop.Should().BeTrue());
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 1 or 3).Should().AllSatisfy(x => x.IsInLoop.Should().BeFalse());
+    }
+
+    [TestMethod]
+    public void Loops_IsInLoop_GoTo()
+    {
+        var code = """
+            Console.WriteLine("Before loop");
+            Start:
+            Console.WriteLine("Inside loop");
+            if (Condition)
+                goto Start;
+            Console.WriteLine("After loop");
+            """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.PostProcessed.GroupBy(x => x.Block.Ordinal).Should().HaveCount(3);
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 2).Should().AllSatisfy(x => x.IsInLoop.Should().BeTrue());
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 1 or 3).Should().AllSatisfy(x => x.IsInLoop.Should().BeFalse());
+    }
```

Comment:
can we add a combination? 
E.g. nested loops or loop and goto.

Also maybe add a counter-example, e.g. a simple if/ternary. 
Because some blocks will be re-visited that way but we want them to NOT be considered `InLoop`